### PR TITLE
Use Setting class to register node.ingest setting

### PR DESCRIPTION
**Pull request by javanna**
Boolean parsing is now strict. Also added isIngestNode methods to DiscoveryNode to align this setting with the existing node.data node.master and node.client. Removed NodeModule#isIngestEnabled methods that are not needed anymore.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/node/NodeModuleTests.java [hunk 1]
```diff
@@ -1,80 +0,0 @@
...
-
-import org.elasticsearch.common.collect.ImmutableOpenMap;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.test.ESTestCase;
-
-import static org.hamcrest.Matchers.equalTo;
-
-public class NodeModuleTests extends ESTestCase {
-
-    public void testIsNodeIngestEnabledSettings() {
```
Comment (martijnvg) [inline core/src/test/java/org/elasticsearch/node/NodeModuleTests.java]: where have these tests been moved to?

Comment (javanna) [inline core/src/test/java/org/elasticsearch/node/NodeModuleTests.java]: they have been deleted :) they turned into something like "lets test that parseBooleanExact works".  which is not needed for each single setting I think.

Comment (martijnvg) [inline core/src/test/java/org/elasticsearch/node/NodeModuleTests.java]: agreed :) this was overkill

**Comment by martijnvg** [comment]
LGTM

**Pull request merged**